packages/rs-drive/src/drive/mod.rs

```diff
@@ -78,9 +80,11 @@ use crate::drive::cache::DriveCache;
 #[cfg(any(feature = "server", feature = "verify"))]
 pub struct Drive {
     /// GroveDB
-    pub grove: GroveDb,
+    pub grove: Arc<GroveDb>,
```

Comment:
Why would we need Arc when GroveDb isn't mutable?

Reply:
`Arc` is not about mutability but about shared ownership via reference counting in multithreaded environment

Shared ownership is needed for HTTP server that serves visualizer app in a separate thread and no one can guarantee that GroveDB instance will outlive the mentioned thread, thus `'static` requirement is needed: either direct ownership of GroveDB for HTTP server which is impossible since drive shall own it, 'static reference which involves heap leaking, or a reference counting, which is used here.

Note that it's actually one Arc for Drive and a weak pointer for visualizer thread, so when drive is dropped then http server will shutdown on attempt to access grovedb, other than that nothing happens regarding counters and it can be treated like a yet another pointer not affecting anything

Reply:
Putting GroveDB behind an Arc will basically slow down the system heavily and is not something I want to do. There should be other ways to do this. We currently share groveDB between threads already I think as GroveDB is just a wrapper around RocksDB @shumkov How is it done for CheckTx again?

Reply:
How it will slow down the system heavily? There will be no atomic operations at all, just an access by pointer;

Yes it can be shared between threads if we use scoped threads, we can't use them for a background task like an HTTP server

Reply:
As I remember performance issues were related only to RwLocks btw

Reply:
@fominok ARC involves atomic operations to increment and decrement the reference count, which introduces overhead. Each clone of an Arc<T> increments the reference count, and each drop decrements it. These atomic operations are slower than simple pointer manipulations involved in borrowing.

Reply:
@QuantumExplorer Exactly, that's why i'm telling there will be no increment/decrement at all after creation, only Arc (1 instance) and Weak (also 1 instance), so no atomic reference counting will happen at all; access to the value wrapped in Arc doesn't involve atomic operations, only Arc::clone and Drop do

A check (that counter is > 0) will happen on Weak side, but it's a Visualizer thread and it won't affect Drive performance

Reply:
I think we already use it with tonic without Arc because Platform itself inside Arc

Reply:
We did benchmarks and this was okay.